### PR TITLE
Remove legacy MetaDataStateFormat

### PR DESCRIPTION
**Pull request by ywelsch**
Removes the legacy (pre-1.5) legacy MetaDataStateFormat (see #10316).

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java [hunk 1]
```diff
@@ -292,39 +286,19 @@ public  T loadLatestState(Logger logger, NamedXContentRegistry namedXContentRegi
...
-                    if (data.length == 0) {
-                        logger.debug("{}: no data for [{}], ignoring...", prefix, stateFile.toAbsolutePath());
-                        continue;
-                    }
-                    try (XContentParser parser = XContentHelper
-                            .createParser(namedXContentRegistry, LoggingDeprecationHandler.INSTANCE, new BytesArray(data))) {
-                        state = fromXContent(parser);
-                    }
-                    if (state == null) {
-                        logger.debug("{}: no data for [{}], ignoring...", prefix, stateFile.toAbsolutePath());
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java]: Except we didn't ignore it, we returned `null` straight away...

**Review by DaveCTurner — APPROVED** [review]
LGTM.

**Comment by DaveCTurner** [comment]
Was a little surprised that no tests needed changing here, and it turns out they do:

https://github.com/elastic/elasticsearch/blob/b7246199db2c788250e03b804493b3548df8b288/server/src/test/java/org/elasticsearch/gateway/MetaDataStateFormatTests.java#L240

**Comment by ywelsch** [comment]
hehe, true

File: server/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java (hunk 1 above)
Comment (ywelsch) [reply server/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java]: details ;)

File: server/src/test/java/org/elasticsearch/bwcompat/RecoveryWithUnsupportedIndicesIT.java [hunk 2]
```diff
@@ -81,16 +81,16 @@ protected Settings prepareBackwardsDataDir(Path backwardsIndex) throws IOExcepti
...
+        String indexName = "unsupported-2.4.5";
 
         logger.info("Checking static index {}", indexName);
         Settings nodeSettings = prepareBackwardsDataDir(getBwcIndicesPath().resolve(indexName + ".zip"));
         try {
             internalCluster().startNode(nodeSettings);
             fail();
         } catch (Exception ex) {
-            assertThat(ex.getCause().getCause().getMessage(), containsString(" was created before v2.0.0.beta1 and wasn't upgraded"));
+            assertThat(ex.getCause().getCause().getMessage(), containsString("Format version is not supported"));
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/bwcompat/RecoveryWithUnsupportedIndicesIT.java]: Could we use `expectThrows()` here?

File: server/src/test/java/org/elasticsearch/bwcompat/RecoveryWithUnsupportedIndicesIT.java [hunk 3]
```diff
@@ -81,16 +81,16 @@ protected Settings prepareBackwardsDataDir(Path backwardsIndex) throws IOExcepti
         return builder.build();
     }
 
-    public void testUpgradeStartClusterOn_0_20_6() throws Exception {
-        String indexName = "unsupported-0.20.6";
+    public void testUpgradeStartClusterOn_2_4_5() throws Exception {
+        String indexName = "unsupported-2.4.5";
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/bwcompat/RecoveryWithUnsupportedIndicesIT.java]: Are any of the other `unsupported-x.x.x.zip` files used?

**Review by DaveCTurner — COMMENTED** [review]
I left a couple of questions.

File: server/src/test/java/org/elasticsearch/bwcompat/RecoveryWithUnsupportedIndicesIT.java (hunk 2 above)
Comment (ywelsch) [reply server/src/test/java/org/elasticsearch/bwcompat/RecoveryWithUnsupportedIndicesIT.java]: sure, fixed in 649f2f2

File: server/src/test/java/org/elasticsearch/bwcompat/RecoveryWithUnsupportedIndicesIT.java (hunk 3 above)
Comment (ywelsch) [reply server/src/test/java/org/elasticsearch/bwcompat/RecoveryWithUnsupportedIndicesIT.java]: not as far as I can see (running the tests will confirm). I initially wanted to remove those in a follow-up, but I think we can do it as well in this PR.

**Comment by ywelsch** [comment]
Can you have another look @DaveCTurner?

**Review by DaveCTurner — APPROVED** [review]
Still LGTM.

**Pull request merged**